framework/include/userobject/NearestPointAverage.h

```diff
@@ -0,0 +1,43 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+// MOOSE includes
+#include "NearestPointBase.h"
+#include "ElementAverageValue.h"
+#include "ElementVariableVectorPostprocessor.h"
+// Forward Declarations
+class NearestPointAverage;
+
+template <>
+InputParameters validParams<NearestPointAverage>();
```

Comment:
this is legacy code. Look at newer objects for how it is done now

Reply:
Ok I think I updated it - let me know if I missed anything